templates/core/Resources/_ti/bootstrap.loader.js

```diff
@@ -0,0 +1,103 @@
+ */
+
+'use strict';
+
+var JS_EXTENSION = '.js',
+	BOOTSTRAP_FILE_NAME = 'bootstrap' + JS_EXTENSION,
+	BOOTSTRAP_SUFFIX = '.' + BOOTSTRAP_FILE_NAME;
+
+exports.loadAsync = function (finished) {
+	var resourceDirectory = Ti.Filesystem.getFile(Ti.Filesystem.getResourcesDirectory()),
+		resourceDirectoryPath = resourceDirectory.nativePath,
+		bootstrapScripts = [];
+
+	// Fetches all "*.bootstrap.js" files under given directory and adds them to the "bootstrapScripts" array.
+	// The names stored in the array can be loaded via require() method.
+	function fetchBootstrapScriptsFrom(file) {
+		var index,
+			fileNameArray,
+			bootstrapPath;
+
+		if (file) {
+			if (file.isDirectory()) {
+				// This is a directory. Recursively look for bootstrap files under it.
+				fileNameArray = file.getDirectoryListing();
+				if (fileNameArray) {
+					for (index = 0; index < fileNameArray.length; index++) {
+						fetchBootstrapScriptsFrom(Ti.Filesystem.getFile(file.nativePath, fileNameArray[index]));
+					}
+				}
+			} else if ((file.name === BOOTSTRAP_FILE_NAME) || file.name.endsWith(BOOTSTRAP_SUFFIX)) {
+				// This is a bootstrap file.
+				// Convert its path to something loadable via require() and add it to the array.
+				bootstrapPath = file.nativePath;
+				bootstrapPath = bootstrapPath.substr(
+					resourceDirectoryPath.length,
+					(bootstrapPath.length - resourceDirectoryPath.length) - JS_EXTENSION.length);
+				bootstrapScripts.push(bootstrapPath);
+			}
+		}
+	}
+	fetchBootstrapScriptsFrom(resourceDirectory);
+
+	// Sort the bootstrap scripts so that they'll be loaded in a consistent order between platforms.
+	bootstrapScripts.sort();
+
+	// Loads all bootstrap scripts found before loading the "app.js" script.
+	function loadBootstrapScripts(finished) {
+		var bootstrapIndex = 0;
+		function doLoad() {
+			// Attempt to load all bootstrap scripts.
+			var fileName, bootstrap;
+			while (bootstrapIndex < bootstrapScripts.length) {
+				// Load the next bootstrap.
+				fileName = bootstrapScripts[bootstrapIndex];
+				bootstrap = require(fileName);
+
+				// Invoke the bootstrap's execute() method if it has one. (This is optional.)
+				// We must wait for the given callback to be invoked before loading the next script.
+				// Note: This is expected to be used to display UI to the end-user.
+				if (bootstrap.execute) {
+					bootstrap.execute(onBootstrapExecutionFinished);
+					return;
+				}
+
+				// We're done with the current bootstrap. Time to load the next one.
+				bootstrapIndex++;
+			}
+
+			// We're done. Invoke given callback so that the "app.js" can be loaded.
+			finished();
+		}
+		function onBootstrapExecutionFinished() {
+			// Last bootstrap has finished execution. Time to load the next one.
+			// Note: Add a tiny delay so whatever UI the last bootstrap loaded has time to close.
+			bootstrapIndex++;
+			setTimeout(function () {
+				doLoad();
+			}, 1);
+		}
+		doLoad();
+	}
+	loadBootstrapScripts(function () {
```

Comment:
This is way too much code and it's too hard to follow. You have scope declared functions and functions that don't need to exist.

A simpler approach would be to do something like the following:

```js
bootstrapScripts.reduce((promise, script) => {
	return promise.then(() => {
		const bootstrap = require(script);
		if (typeof bootstrap.execute === 'function') {
			return new Promise(resolve => bootstrap.execute(() => setTimeout(resolve, 1)));
		}
	});
}, Promise.resolve());
```

That reduces 42 lines of code down to 8 and that's a win.